dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/plugins/server/servlet/ServletUtil.java

```diff
@@ -0,0 +1,179 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.jboss.resteasy.plugins.server.servlet;
+
+import org.jboss.resteasy.core.Headers;
+import org.jboss.resteasy.specimpl.ResteasyHttpHeaders;
+import org.jboss.resteasy.specimpl.ResteasyUriInfo;
+import org.jboss.resteasy.specimpl.ResteasyUriInfo.InitData;
+import org.jboss.resteasy.util.HttpHeaderNames;
+import org.jboss.resteasy.util.MediaTypeHelper;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Cookie;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
+ * @version $Revision: 1 $
+ */
+public class ServletUtil
+{
+   private static Map<String, InitData> initDataCache = new HashMap<String, InitData>();      // Liberty change
```

Comment:
HashMap is not thread safe.  ConcurrentHashMap?  Any concerns about this causing a leak?

Reply:
I can switch to CHM, probably the perf is not noticeably different. Was not much concerned about thread safety since this map will be write-once/read-many, no mods or dels of entries.

Thought about whether this might cause a leak, too, concluded that it was unlikely. There will be one entry in the map for every <scheme><server><port><uri>  combination: <scheme><port> will be paired usually (one http port, one https port) and <server> will be singular, so basically two entries per URI for configs allowing both http and https access to an app. The number of URIs is finite, and usually small.